uefi-hello-world-aarch64/build.py

```diff
@@ -0,0 +1,396 @@
+
+    xbuild_args = [
+        '--package', 'uefi-hello-world-aarch64',
+        *test_flags,
+    ]
+
+    if SETTINGS['config'] == 'release':
+        xbuild_args.append('--release')
+
+    run_xbuild(*xbuild_args)
+
+    # Copy the built test runner file to the right directory for running tests.
+    built_file = build_dir() / 'uefi-hello-world-aarch64.efi'
+
+    boot_dir = esp_dir() / 'EFI' / 'Boot'
+    boot_dir.mkdir(parents=True, exist_ok=True)
+
+    output_file = boot_dir / f'Boot{SETTINGS["efi_prefix"]}.efi'
+
+    shutil.copy2(built_file, output_file)
+
+
+def clippy():
+    'Runs Clippy on all projects'
+
+    run_xclippy('--all')
+
+
+def doc():
+    'Generates documentation for the library crates.'
+    sp.run([
+        'cargo', 'doc', '--no-deps',
+        '--package', 'uefi',
+        '--package', 'uefi-exts',
+        '--package', 'uefi-alloc',
+        '--package', 'uefi-logger',
+        '--package', 'uefi-services',
+    ])
+
+
+def ovmf_files(ovmf_dir):
+    'Returns the tuple of paths to OVMF_CODE.fd and OVMF_VARS.fd given the directory'
+    return ovmf_dir / SETTINGS['ovmf']['firmware']['code'], ovmf_dir / SETTINGS['ovmf']['firmware']['vars']
+
+
+def check_ovmf_dir(ovmf_dir):
+    'Check whether the given directory contains necessary OVMF files'
+    ovmf_code, ovmf_vars = ovmf_files(ovmf_dir)
+    return ovmf_code.is_file() and ovmf_vars.is_file()
+
+
+def find_ovmf():
+    'Find path to OVMF files'
+
+    # If the path is specified in the settings, use it.
+    if SETTINGS['ovmf']['dir'] is not None:
+        ovmf_dir = SETTINGS['ovmf']['dir']
+        if check_ovmf_dir(ovmf_dir):
+            return ovmf_dir
+        raise FileNotFoundError(f'OVMF files not found in `{ovmf_dir}`')
+
+    # Check whether the test runner directory contains the files.
+    ovmf_dir = WORKSPACE_DIR / f'uefi-hello-world-{ARCH}'
+    if check_ovmf_dir(ovmf_dir):
+        return ovmf_dir
+
+    if sys.platform.startswith('linux'):
+        possible_paths = [
+            # Most distros, including CentOS, Fedora, Debian, and Ubuntu.
+            Path('/usr/share/OVMF'),
+            # Arch Linux
+            Path('/usr/share/ovmf/x64'),
+        ]
+        for path in possible_paths:
+            if check_ovmf_dir(path):
+                return path
+
+    raise FileNotFoundError(f'OVMF files not found anywhere')
+
+
+def run_qemu():
+    'Runs the code in QEMU.'
+
+    # Rebuild all the changes.
+    build('--features', 'qemu')
+
+    ovmf_code, ovmf_vars = ovmf_files(find_ovmf())
+
+    examples_dir = build_dir() / 'examples'
+
+    qemu_monitor_pipe = 'qemu-monitor'
+
+    qemu_flags = [
+        # Disable default devices.
+        # QEMU by defaults enables a ton of devices which slow down boot.
+        # '-nodefaults',
+
+        # Use a generic ARM environment. Sadly qemu can't emulate a RPi 4 like machine though
+        '-machine', 'virt',
+
+        # a57 is a very generic 64-bit ARM CPU in the wild
```

Comment:
whoops would you like to change the errata from a57 to a72?

Reply:
We'll probably want to merge this build script back into the original one eventually, we can fix the comment at that point.